restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/client/api/CallApiClient.java

```diff
@@ -0,0 +1,163 @@
+ *
+ */
+package org.restcomm.connect.http.client.api;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.text.ParseException;
+import java.util.ArrayList;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.Header;
+import org.apache.http.HttpHeaders;
+import org.apache.http.NameValuePair;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.message.BasicNameValuePair;
+import org.restcomm.connect.commons.dao.Sid;
+import org.restcomm.connect.commons.faulttolerance.RestcommUntypedActor;
+import org.restcomm.connect.commons.util.UriUtils;
+import org.restcomm.connect.dao.DaoManager;
+import org.restcomm.connect.dao.entities.Account;
+import org.restcomm.connect.dao.entities.CallDetailRecord;
+import org.restcomm.connect.http.asyncclient.HttpAsycClientHelper;
+import org.restcomm.connect.http.client.DownloaderResponse;
+import org.restcomm.connect.http.client.HttpRequestDescriptor;
+import org.restcomm.connect.telephony.api.Hangup;
+
+import akka.actor.ActorRef;
+import akka.actor.Props;
+import akka.actor.UntypedActor;
+import akka.actor.UntypedActorFactory;
+import akka.event.Logging;
+import akka.event.LoggingAdapter;
+
+public class CallApiClient extends RestcommUntypedActor {
+
+    private final LoggingAdapter logger = Logging.getLogger(getContext().system(), this);
+
+    private final DaoManager storage;
+    private final Sid callSid;
+    private CallDetailRecord callDetailRecord;
+    private static final String SUPER_ADMIN_ACCOUNT_SID="ACae6e420f425248d6a26948c17a9e2acf";
+    private ActorRef requestee;
+
+    private ActorRef httpAsycClientHelper;
+
+    public CallApiClient(final Sid callSid, final DaoManager storage) {
+        super();
+        this.callSid = callSid;
+        this.storage = storage;
+    }
+    @Override
+    public void onReceive(Object message) throws Exception {
+        final Class<?> klass = message.getClass();
+        final ActorRef sender = sender();
+        ActorRef self = self();
+
+        if (logger.isInfoEnabled()) {
+            logger.info(" ********** CallApiClient " + self().path() + " Sender: " + sender);
+            logger.info(" ********** CallApiClient " + self().path() + " Processing Message: " + klass.getName());
+        }
+        if (Hangup.class.equals(klass)) {
+            onHangup((Hangup) message, self, sender);
+        } else if (DownloaderResponse.class.equals(klass)) {
+            onDownloaderResponse(message, self, sender);
+        }
+    }
+
+    private void onDownloaderResponse(Object message, ActorRef self, ActorRef sender) {
+        final DownloaderResponse response = (DownloaderResponse) message;
+        if (logger.isInfoEnabled()) {
+            logger.info("Call api response succeeded " + response.succeeded());
+            logger.info("Call api response: " + response);
+        }
+        requestee.tell(message, self);
```

Comment:
@maria-farooq with the current implementation the `requestee` will always be null. 
Shouldn't we protect the code for this?

Reply:
please chek comments below